pkg/network/ebpf/c/runtime/tracer.c

```diff
@@ -398,6 +398,11 @@ int kprobe__tcp_set_state(struct pt_regs* ctx) {
     tcp_stats_t stats = { .state_transitions = (1 << state) };
     update_tcp_stats(&t, stats);
 
+    __u8 old_state = 0;
+    bpf_probe_read(&old_state, sizeof(old_state), (void *)&sk->sk_state);
+    if (old_state == TCP_SYN_SENT)
```

Comment:
Do we need to know the previous state? Could we do this for both directions and remove the need to check the previous state?

Reply:
It would indeed work without the check, and that's what I tried originally. However, we do already report server-side connection elsewhere (probably from our hook on `inet_csk_accept`), so we end up with two reported connections server-side when there is only one.

Reply:
You can't really use the full tuple here (with the PID) likely because the pid is not available (or is different from the actual app PID, likely a kthread). So only the tcp stats can be updated here.

Reply:
Actually, the two PIDs are respectively the client and the server processes.